Adjusted schemas

cohnjesse opened the pull request:
## Description
Original ticket - https://github.com/department-of-veterans-affairs/va.gov-team/issues/4640

In order to get the form 686 work from the Rainbows team working on staging we need to fix the form's submitted data structure so that it both matches the shape of the JSON schema for the back end AND is valid for the VA form system.

## Testing done
Unit tests will be included in a separate branch

## Acceptance criteria
- [x] The shape of the form data on submission matches the shape of the JSON schema object for the back end
- [x] No schema invalidations errors get thrown when clicking the submit button

cohnjesse committed:
Adjusted schemas

cohnjesse committed:
simplified locationSchema, removed genericLocation

cohnjesse committed:
adjusted for prettier

micahchiang commented on src/applications/disability-benefits/686/config/form.js:
```diff
@@ -69,6 +69,7 @@ const {
   domesticAddress,
   fullName,
   internationalAddressText,
+  genericLocation,
```
Should this be removed? I was under the assumption we were getting rid of it moving forward.

micahchiang commented on src/applications/disability-benefits/686/config/form.js:
```diff
@@ -269,41 +277,16 @@ function createLocationUISchemaForKey(
-      'ui:title': 'Country',
+    state: {
+      'ui:title': 'State (or country if outside the USA',
       'ui:required': isRequiredCallback,
```
Can this property be removed if we're always asking for city and state (or country) moving forward?

micahchiang commented on src/applications/disability-benefits/686/config/form.js:
```diff
@@ -269,41 +277,16 @@ function createLocationUISchemaForKey(
-          !isUSAAddress(get(`${insertRealIndexInKey(key, index)}`, formData)),
-      },
+      'ui:title': 'City or county', 
+      'ui:required': isRequiredCallback,
```
Can this property be removed if we're always asking for city and state (or country) moving forward?

micahchiang commented on src/platform/forms-system/src/js/validation.js:
```diff
@@ -267,8 +267,14 @@ export function isValidForm(form, pageList) {
 
+      console.log(result);
+      console.log();
+
```
These console logs should be removed.

micahchiang reviewed (changes requested):
Some questions about a couple of the properties, and also a comment to remove the console logs in `validation.js`.

cohnjesse committed:
removed console logs

cohnjesse committed:
Removing test as back end is failing

the back end endpoint for this is not finished so it is failing. Removing for now until the back end is fixed, then we can add it back.

cohnjesse committed:
Fixed unit test DOM references

These tests were referencing the old form config with less fields, updated them to the new numbers of fields

cohnjesse committed:
Added inline form properties

These are being added here so that we can change them without affecting the schema files

cohnjesse committed:
Adjusted unit tests to match field numbers

The number of fields in the form changed so needed to update the unit tests to make sure they have the correct numbers

cohnjesse committed:
Adjusted based on linter

cohnjesse committed:
rerun build

micahchiang commented on src/applications/disability-benefits/686/config/form.js:
```diff
@@ -269,40 +274,13 @@ function createLocationUISchemaForKey(
-    countryDropdown: {
-      'ui:title': 'Country',
+    state: {
+      'ui:title': 'State (or country if outside the USA',
```
I think this is missing an ending paren ')' after USA.

micahchiang reviewed (approved):
LGTM

cohnjesse committed:
Merge branch 'master' into 4640-resolve-JSON-issues

cohnjesse committed:
added missing param
